src/weights.jl

```diff
@@ -650,3 +650,24 @@ wquantile(v::RealVector, w::AbstractWeights{<:Real}, p::RealVector) = quantile(v
 wquantile(v::RealVector, w::AbstractWeights{<:Real}, p::Number) = quantile(v, w, [p])[1]
 wquantile(v::RealVector, w::RealVector, p::RealVector) = quantile(v, weights(w), p)
 wquantile(v::RealVector, w::RealVector, p::Number) = quantile(v, weights(w), [p])[1]
+
+
+"""
+    wiqr(v, w)
+
+Compute the interquartile range (IQR) of an array with weights `w`, given as either a vector
+or an `AbstractWeights` vector.
+"""
+wiqr{W <: Real}(v::RealVector,w::AbstractWeights{W}) = (q = quantile(v, w, [.25, .75]); q[2] - q[1])
+wiqr(v::RealVector,w::RealVector) = (q = quantile(v, weights(w), [.25, .75]); q[2] - q[1])
+
+
+"""
+    iqr(v, w)
+
+Compute the interquartile range (IQR) of an array with weights `w`, given as either a vector
```

Comment:
I'd rather require `w` to be `AbstractWeights`, as we do for other methods. Then instead of mentioning it in the text, add it to the signature.

BTW, could you also give the definition of the IQR?

Reply:
`quantile` and `wquantile` both have methods to accept `w` as a `RealVector` (which gets converted to general `Weights`) in addition to the methods where `w` is passed as `AbstractWeights`. I just followed that convention. Would you still like me to change it?

Reply:
Are you sure? AFAICT only `wquantile` accepts a `RealVector` (which sounds fine and `wiqr` can do the same thing).

Reply:
Ah, subtle differences. I just glanced over that. Ok I'll make these changes on Monday.